mmaction/models/backbones/vit_mae.py

```diff
@@ -0,0 +1,374 @@
+    """Generate sinusoid encoding table.
+
+    Sinusoid encoding is a kind of relative position encoding method came from
+    `Attention Is All You Need<https://arxiv.org/abs/1706.03762>`_.
+    Args:
+        n_position (int): The length of the input token.
+        embed_dims (int): The position embedding dimension.
+    Returns:
+        :obj:`torch.FloatTensor`: The sinusoid encoding table.
+    """
+
+    vec = torch.arange(embed_dims, dtype=torch.float64)
+    vec = (vec - vec % 2) / embed_dims
+    vec = torch.pow(10000, -vec).view(1, -1)
+
+    sinusoid_table = torch.arange(n_position).view(-1, 1) * vec
+    sinusoid_table[:, 0::2].sin_()  # dim 2i
+    sinusoid_table[:, 1::2].cos_()  # dim 2i+1
+
+    sinusoid_table = sinusoid_table.to(torch.float32)
+
+    return sinusoid_table.unsqueeze(0)
+
+
+@MODELS.register_module()
+class VisionTransformer(BaseModule):
+    """Vision Transformer with support for patch or hybrid CNN input stage. An
+    impl of `VideoMAE: Masked Autoencoders are Data-Efficient Learners for
+    Self-Supervised Video Pre-Training <https://arxiv.org/pdf/2203.12602.pdf>`_
+
+    Args:
+        img_size (int | tuple): Size of input image. Defaults to 224.
+        patch_size (int): Spatial size of one patch. Defaults to 16.
+        in_channels (int): The number of channels of he input. Defaults to 3.
+        embed_dims (int): Dimensions of embedding. Defaults to 768.
+        depth (int): number of blocks in the transformer. Defaults to 12.
+        num_heads (int): Number of parallel attention heads in
+            TransformerCoder. Defaults to 12.
+        mlp_ratio (int): The ratio between the hidden layer and the input layer
+            in the FFN. Defaults to 4.
+        qkv_bias (bool): If True, add a learnable bias to q and v.
+            Defaults to True.
+        qk_scale (float | optional): Override default qk scale of
+            ``head_dim ** -0.5`` if set. Defaults to None.
+        drop_rate (float | optional): Dropout ratio of output. Defaults to 0.
+        attn_drop_rate (float): Dropout ratio of attention weight.
+            Defaults to 0.
+        drop_path_rate (float): Dropout ratio of the residual branch.
+            Defaults to 0.
+        norm_cfg (dict): Config for norm layers. Defaults to
+            `dict(type='LN', eps=1e-6)`.
+        init_values (float): Value to init the multiplier of the residual
+            branch. Defaults to 0.
+        use_learnable_pos_emb (bool): If True, use learnable positional
+            embedding, othersize use sinusoid encoding. Defaults to False.
+        num_frames (int): Number of frames in the video. Defaults to 16.
+        tubelet_size (int): Temporal size of one patch. Defaults to 2.
+        use_mean_pooling (bool): If True, take the mean pooling over all
+            positions. Defaults to True.
+        init_cfg (dict | None): The Config for initialization. Defaults to
+            None.
```

Comment:
```suggestion
        init_cfg (Union[dict, :obj:`mmengine.ConfigDict`], optional): The
            config for initialization. Defaults to None.
```

Reply:
It should be dict or ConfigDict as suggested in mmengine